### PR TITLE
macOS support

### PR DESCRIPTION
**Pull request by lukabratos**
### What?

Add support for macOS.

screen shot 2018-01-24 at 14 54 56

----
CC @pusher/mobile

### CODE REVIEW

File: PushNotifications.podspec [hunk 1]
```diff
@@ -7,9 +7,11 @@ Pod::Spec.new do |s|
   s.author           = { "Luka Bratos" => "luka@pusher.com" }
   s.source           = { git: "https://github.com/pusher/push-notifications-swift.git", tag: s.version.to_s }
   s.social_media_url = 'https://twitter.com/pusher'
+  s.documentation_url = ''
```
Comment (hamchapman) [inline PushNotifications.podspec]: Is this needed?

File: Sources/PushNotifications.swift [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
+#endif
+import Foundation
 
 @objc public final class PushNotifications: NSObject {
     private var deviceId: String?
     private var instanceId: String?
     private var baseURL: String?
     private let session = URLSession.shared
-
+
```
Comment (hamchapman) [inline Sources/PushNotifications.swift]: Probs don't want this

File: Sources/PushNotifications.swift [hunk 3]
```diff
@@ -188,17 +201,23 @@ import UserNotifications
...
+    #if os(iOS)
     private func registerForPushNotifications(application: UIApplication) {
         UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .sound, .badge]) { (granted, error) in
             if (granted) {
                 DispatchQueue.main.async {
                     application.registerForRemoteNotifications()
                 }
             }
-
+
```
Comment (hamchapman) [inline Sources/PushNotifications.swift]: Probs don't want this

File: Sources/PushNotifications.swift [hunk 4]
```diff
@@ -188,17 +201,23 @@ import UserNotifications
...
-
+            
             if let error = error {
                 print(error.localizedDescription)
             }
         }
     }
+    #elseif os(OSX)
+    private func registerForPushNotifications(application: NSApplication) {
+        application.registerForRemoteNotifications(matching: [.alert, .sound, .badge])
```
Comment (hamchapman) [inline Sources/PushNotifications.swift]: More of a question than anything - are we assuming that every user will want to register for all of `[.alert, .sound, .badge]` notifications?

File: push-notifications-ios/push-notifications-ios/AppDelegate.swift [hunk 5]
```diff
@@ -3,16 +3,16 @@ import PushNotifications
 
 @UIApplicationMain
 class AppDelegate: UIResponder, UIApplicationDelegate {
-
+
```
Comment (hamchapman) [inline push-notifications-ios/push-notifications-ios/AppDelegate.swift]: Again, more whitespace changes in here that you probably don't want

File: PushNotifications.podspec (hunk 1 above)
Comment (lukabratos) [reply PushNotifications.podspec]: This sneaked in unintentionally. I wanted to do a separate PR for this.
But if it's in already now I might as well put a link to the docs in 👍

File: Sources/PushNotifications.swift (hunk 4 above)
Comment (lukabratos) [reply Sources/PushNotifications.swift]: If will be more flexible in the future. Users will be able to provide an array with options so we will register just specified notifications. It's on TODO 👍

**Comment by lukabratos** [comment]
SysInfo is a helper which enables me to identify the platform (iOS or macOS). I've used it in [PushNotifications.h](https://github.com/pusher/push-notifications-swift/pull/24/files#diff-a86d9bbaeda5f1e71f5df0dd1b6a55ab) so I know if I need to import Cocoa or UIKit.

**Comment by hamchapman** [comment]
Interesting - can you not use `#if os(iOS)` and `#if os(OSX)` without the SysInfo stuff? I think that's what we are doing in pusher-websocket-swift at least.

Maybe related, what do you need Cocoa or UIKit for?

**Comment by lukabratos** [comment]
Actually, you're right. `#import <UIKit/UIKit.h>` was imported automatically when the header was generated by Xcode.
I've looked at [PusherSwift](https://github.com/pusher/pusher-websocket-swift/blob/master/Sources/PusherSwift.h) and replaced UIKit with Foundation + removed SysInfo!

Thanks for flagging this 💯

**Pull request merged**